src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupService.scala

```diff
@@ -0,0 +1,88 @@
+      _ <- createPolicyForMembers(managedGroup)
+      policies <- resourceService.accessPolicyDAO.listAccessPolicies(managedGroup)
+      _ <- createAggregateGroup(managedGroup, policies)
+    } yield managedGroup
+  }
+
+  private def createPolicyForMembers(managedGroup: Resource): Future[AccessPolicy] = {
+    val accessPolicyName = AccessPolicyName(memberRole.roleName.value)
+    val resourceAndPolicyName = ResourceAndPolicyName(managedGroup, accessPolicyName)
+    resourceService.createPolicy(resourceAndPolicyName, members = Set.empty, Set(memberRole.roleName), actions = Set.empty)
+  }
+
+  private def createAggregateGroup(resource: Resource, componentPolicies: Set[AccessPolicy]): Future[BasicWorkbenchGroup] = {
+    val email = generateManagedGroupEmail(resource.resourceId)
+    val workbenchGroupName = WorkbenchGroupName(resource.resourceId.value)
+    val groupMembers: Set[WorkbenchSubject] = componentPolicies.map(_.id)
+    resourceService.directoryDAO.createGroup(BasicWorkbenchGroup(workbenchGroupName, groupMembers, email))
+  }
+
+  private def generateManagedGroupEmail(resourceId: ResourceId): WorkbenchEmail = {
+    val groupName = resourceId.value
+    validateGroupName(groupName)
+    WorkbenchEmail(constructEmail(groupName))
+  }
+
+  private def constructEmail(groupName: String) = {
+    s"${groupName}@${resourceService.emailDomain}"
+  }
+
+  private def validateGroupName(groupName: String) = {
+    val errors = validateGroupNamePattern(groupName) ++ validateGroupNameLength(constructEmail(groupName))
+    if (errors.nonEmpty)
+      throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "Cannot create valid email address with the group name provided" , errors.toSeq))
+  }
+
+  private def validateGroupNamePattern(str: String): Option[ErrorReport] = {
+    str match {
+      case ManagedGroupService.GroupNameRe() => None
+      case _ => Option(ErrorReport(s"You have specified a group name that contains characters that are not permitted in an email address. Group name may only contain alphanumeric characters, underscores, and dashes"))
+    }
+  }
+
+  private val maxLength = 64
+  private def validateGroupNameLength(str: String): Option[ErrorReport] = {
+    if (str.length >= maxLength)
+      Option(ErrorReport(s"Email address '$str' is ${str.length} characters in length.  Email address length must be shorter than $maxLength characters"))
+    else
+      None
+  }
+
+  // Per dvoet, when asking for a group, we will just return the group email
+  def loadManagedGroup(groupId: ResourceId): Future[Option[WorkbenchEmail]] = {
+    resourceService.directoryDAO.loadGroup(WorkbenchGroupName(groupId.value)).map(_.map(_.email))
+  }
+
+  def deleteManagedGroup(groupId: ResourceId) = {
+    resourceService.directoryDAO.deleteGroup(WorkbenchGroupName(groupId.value))
+    resourceService.deleteResource(Resource(managedGroupType.name, groupId))
+  }
+}
+
+object ManagedGroupService {
```

Comment:
Not sure where that other thread went, but I'll chime in and vote for initial lowercase as well. `val`s are immutable and are far more common than `var`s in scala so there isn't any need to call special attention to these. My brain is already rewired to read the initial uppercase as a type name rather than a val name.

Reply:
fixed